src/nvim/mbyte.c

```diff
@@ -555,6 +555,24 @@ size_t mb_string2cells(const char_u *str)
   return clen;
 }
 
+/// Get the number of cells occupied by string `str` with maximum length `size`
+///
+/// @param str The source string, may not be NULL, must be a NUL-terminated
+///            string.
+/// @param size maximum length of string. It will terminate on earlier NUL.
+/// @return The number of cells occupied by string `str`
+size_t mb_string2cells_len(const char_u *str, size_t size)
```

Comment:
shouldn't new functions use `char *` ?

Reply:
That would make it inconsistent from other mbyte functions, as well as causing a lot of unnecessary casts between different mbyte functions. If making mbyte functions `char *` is a goal, it would seem better with a dedicated PR for this with a well though out design to minimise internal casts as well as reducing the risk of semantic errors.